pkgs/by-name/en/encore/go.nix

```
@@ -0,0 +1,87 @@
+{ fetchFromGitHub
+, stdenv
+, lib
+, go
+, ...
+}:
+stdenv.mkDerivation rec {
+  pname = "go-encore";
+  version = "1.22";
+
+  src = fetchFromGitHub {
+    owner = "encoredev";
+    repo = "go";
+    rev = "encore-go${version}";
+    sha256 = "sha256-SSa3CoUS6JMfs6T1PMb3NK8UeJnpLlTIn46/3oCTnL8=";
+  };
+
+  buildInputs = [ go ];
+
+  patchPhase =
+    let
+      goSrc = fetchFromGitHub {
+        owner = "golang";
+        repo = "go";
+        rev = "release-branch.go${version}";
+        hash = "sha256-amASGhvBcW90dylwFRC2Uj4kOAOKCgWmFKhLnA9dOgg=";
+      };
+    in
+    ''
+      # delete submodule and replace with the go source
+      rm -rf go
+      cp -r ${goSrc} go
+      chmod -R u+rw go
+
+      cd go
```

Would using pushd/popd convey the intent more clearly?
```suggestion
      pushd go
```